packages/@react-spectrum/datepicker/test/DatePickerBase.test.js

```diff
@@ -492,6 +462,74 @@ describe('DatePickerBase', function () {
       let segments = getAllByRole('spinbutton');
       expect(segments[0]).toHaveFocus();
     });
+
+    describe('RTL focus management', function () { 
+      beforeEach(() => {
+        jest.spyOn(HTMLElement.prototype, 'getBoundingClientRect').mockImplementation(function () {
+          let x = 0;
+          let el = this;
+  
+          if (el.getAttribute('role') === 'spinbutton') {
+            if (el.parentElement.getAttribute('data-testid') === 'end-date') {
```

Comment:
i thought the issue with some of these was that the dom order was not in alignment with the rendered order?
this test is a good base, but should we include some tests where the order differs? I think you could use `el.getAttribute('data-type')` to order based on the day/month/hour/etc

Reply:
ah yes, it's more obvious with when using examples with time segments so i should update the examples to include those